### ISSUE TITLE
getInvoices with pager does not include Tracking in LineItems

### ISSUE THREAD

**Issue by Maestro501** [XeroAPI/xero-node#22]
Node version: 6.10.2
npm version: 3.10.10

What I did
```
        xeroClient.core.invoices.getInvoices({
          where: 'Status!="PAID" AND Status!="VOIDED" AND Type=="ACCREC"',
          pager: {
            start: 1,
            callback: (err, response, cb) => {
              if (err) { throw err; }
              cb(response.data);
            },
          },
        })
```

What I got
```
[... 
  {... 
     LineItems: [...
       {...
         Tracking: [{}]
       }
     ]
  }
]
```

What I get on [Xero API Preview](https://app.xero.com/Preview/invoices) for same request & invoice & line item
```
<Tracking>
    <TrackingCategory>
        <Name>Customer Contract</Name>
        <Option>***</Option>
        <TrackingCategoryID>***</TrackingCategoryID>
    </TrackingCategory>
    <TrackingCategory>
        <Name>Department</Name>
        <Option>Engineering</Option>
        <TrackingCategoryID>***</TrackingCategoryID>
    </TrackingCategory>
</Tracking>
```

**Comment by jordanwalsh23** [comment]
Thanks for raising.

This is a known limitation at the moment. Feel free to fork the code and
submit a PR.

Alternatively I'll get round to this eventually.

If you need this as part of the Xero partner program please submit a ticket
to api@xero.com and this may get some more attention.

**Comment by Maestro501** [comment]
Thanks for the followup.

No, this is not part of the Xero partner program yet, just in ideation phase for it.

I'll try see if I can find some time for a PR.
Do you have any rough idea where the limitation stems from?

**Comment by jordanwalsh23** [comment]
Yes. It'll be the fromXmlObject function in the entities/invoice.js schema.

It won't be serialising the data back from XML correctly so those arrays
will be blank.

Shouldn't be too hard to fix. But needs a bit of effort around testing.

**Comment by Maestro501** [comment]
I get `Error: Cannot find module './config/testing_config.json'` when trying to run
```
git clone ..
npm install
npm test
```

this file seems to be ignored from git as well..
Any solution to this?

**Comment by jordanwalsh23** [comment]
In order to test you need to have a testing_config.json set up.

Check the example_test_config.json file, modify your parameters to suit,
then save as testing_config.json.

Run npm test again and this should work.

**Comment by Maestro501** [comment]
Deleted my previous comment because I found your sample file =P 
Didn't think you'd reply so quickly.

**Comment by Maestro501** [comment]
So I managed to get the tests to run, but I don't think the current test logic cleans up all the invoices the test creates.. So I had to go to my Xero to manually delete all of them..

I noticed in the test file that some of the resources do actually get cleaned up through `after` hook (for example, bankAccounts) though.

Is the test intended to clean up for itself? (in which case, I guess a PR to clean up the invoices should happen first) or Am I not supposed to be using my actual Xero account credentials when contributing to this project?

Is there a sandbox environment, or a way for me to create the credentials simply for contributing to this project, instead of tempering with my actual Xero account for my business?

Thanks in advance.

**Comment by jordanwalsh23** [comment]
Definitely do not use a real account when running the tests! There's no
guarantee that it will self Clean.

Xero provides a free 'Demo Org' with every subscription. You can enable
this from your My Xero menu footer.

Please use this so it won't matter what happens to the file.

**Comment by jordanwalsh23** [comment]
Hey @Maestro501, how did you go on this problem? I'm looking to fix it in an upcoming release, but if you've already got a solution you could save me some time.  

Let me know.

**Comment by jordanwalsh23** [comment]
I've just pushed PR #89 for review. Once @philals has some time to review we'll get this merged which should resolve this issue.

**Comment by jordanwalsh23** [comment]
Looks like this change has been merged.  Awaiting the next release to NPM prior to closing.

**Comment by philals** [comment]
Hi @Maestro501 I have merged @jordanwalsh23 PR #89 and it's on NPM as 2.7.0.

Let us know if you have any more problems.

Thanks
